src/models/iceflow/SIA2D/SIA2D_utils.jl

```diff
@@ -266,4 +266,40 @@ function surface_V(H::Matrix{F}, simulation::SIM) where {F <: AbstractFloat, SIM
     Vy = - D .* avg_x(dSdy)
 
     return Vx, Vy    
-end
\ No newline at end of file
+end
+
+function H_from_V(V::Matrix{<:Real}, simulation::SIM) where {SIM <: Simulation}
+    params::Sleipnir.Parameters = simulation.parameters
+    
+    iceflow_model = simulation.model.iceflow
+    glacier::Sleipnir.Glacier2D = simulation.glaciers[iceflow_model.glacier_idx[]]
+    B = glacier.B
+    Δx = glacier.Δx
+    Δy = glacier.Δy
+    A = iceflow_model.A
+    n = iceflow_model.n
+    C = iceflow_model.C
+    ρ = params.physical.ρ
+    g = params.physical.g
+    H₀ = glacier.H₀
+
+    # Update glacier surface altimetry
+    S = glacier.S  
+    V = Huginn.avg(V)
+    
+    # All grid variables computed in a staggered grid
+    # Compute surface gradients on edges
+    dSdx = Huginn.diff_x(S) / Δx
+    dSdy = Huginn.diff_y(S) / Δy
+    ∇S = (Huginn.avg_y(dSdx).^2 .+ Huginn.avg_x(dSdy).^2).^(1/2)
+    ∇S[V .== 0] .= 0
```

Comment:
Why do you need this?

Reply:
Well setting `S` to zero at where the velocity is zero was done to avoid some computation issue, but I can't remember what the issue was exactly. But I am very sure that it was necessary.

Reply:
Okay never mind I double checked and it is redundant.